docs/API.md

```diff
@@ -327,7 +327,7 @@ Options can be configured as a parameter to the [init()](./interfaces/tracer.htm
 | runtimeMetrics | `DD_RUNTIME_METRICS_ENABLED`   | `false`     | Whether to enable capturing runtime metrics. Port 8125 (or configured with `dogstatsd.port`) must be opened on the agent for UDP. |
 | reportHostname | `DD_TRACE_REPORT_HOSTNAME`     | `false`     | Whether to report the system's hostname for each trace. When disabled, the hostname of the agent will be used instead. |
 | experimental   | -                              | `{}`        | Experimental features can be enabled all at once using boolean `true` or individually using key/value pairs. Please contact us to learn more about the available experimental features. |
-| plugins        | -                              | `true`      | Whether or not to enable automatic instrumentation of external libraries using the built-in plugins. |
+| plugins        | `DD_TRACE_DISABLED_PLUGINS`    | `true`      | Whether or not to enable automatic instrumentation of external libraries using the built-in plugins. Either a boolean or an Array of integration names to automatically disable when the tracer is initialized e.g. `['express', 'dns']`. When passed an en environment variable, the format is a comma seperated string of integration names to automaticaly disable, e.g. `'express,http,dns'`|
```

Comment:
Let's remove the boolean as this is something we are thinking to deprecate, and in general it might make more sense as another header if we decide to keep it. The syntax for the array should also be CSV which is generally more intuitive for a header.

Reply:
updated the `disabledPlugins` syntax to be csv only. I am a little bit confused on your note here:

>  `Let's remove the boolean as this is something we are thinking to deprecate, and in general it might make more sense as another header if we decide to keep it.` 

as per your other comment, we're splitting out `disabledPlugins` into a different env var (which will not have a boolean config option). Are you saying you'd like me to deprecate `plugins` in this PR as well? I'm not 100% sure on the deprecation strategy for this repo, do we generally log a warning for a few versions before removing a feature? If you'd like I can work on deprecating/removing `plugins` as a config option entirely but might be better to split that into a different PR since the scope is a bit different than what's happening here? Lmk your thoughts, apols if i've misunderstood intent here.

Reply:
What I'm actually thinking is to have just `plugins` as an in-code configuration option, with no change compared with today, and the new environment variable to disable the plugins. So really all that would be added is the new environment variable. While this is a bit more difficult to document, it allows to not have conflicting configuration options when we add an object config for `plugins`.

Reply:
ah, gotcha, that certainly makes implementation easier. I think as long as the documentation is clear it's fine, given that longer term a more general `plugins` config option will cover this use case, and users can always set the env var via something like `process.env.DD_TRACE_DISABLED_PLUGINS='x,y,z'` if they insist on doing it via application code